includes/class-woocommerce.php

```diff
@@ -227,6 +227,30 @@ private function define( $name, $value ) {
 		}
 	}
 
+	/**
+	 * Returns true if the request is a non-legacy REST API request.
+	 *
+	 * Legacy REST requests should still run some extra code for backwards compatibility.
+	 *
+	 * @todo: replace this function once core WP function is available: https://core.trac.wordpress.org/ticket/42061.
+	 *
+	 * @return bool
+	 */
+	private static function is_rest_api_request() {
+		$request_uri = $_SERVER['REQUEST_URI']; // @codingStandardsIgnoreLine
+		if ( empty( $request_uri ) ) {
+			return false;
+		}
+
+		// REST API prefix.
+		$rest_prefix = trailingslashit( rest_get_url_prefix() );
+
+		// Check if this is a WC endpoint.
+		$is_woocommerce_endpoint = ( false !== strpos( $request_uri, $rest_prefix . 'wc/' ) );
+
+		return apply_filters( 'woocommerce_is_request_to_nonlegacy_rest_api', $is_woocommerce_endpoint );
```

Comment:
Not a fan of this filter name. I understood you used it to avoid conflict with an existing filter (woocommerce_rest_is_request_to_rest_api).

Why not `woocommerce_is_rest_api_request`? 

I'd suggest using this new method in `WC_REST_Authentication::is_request_to_rest_api` in additon to the 3rd party check there so the filter is ran when needed.

Reply:
👍